x/valset-pref/validator_set.go

```diff
@@ -302,14 +302,38 @@ func (k Keeper) getValAddrAndVal(ctx sdk.Context, valOperAddress string) (sdk.Va
 }
 
 // IsPreferenceValid loops through the validator preferences and checks its existence and validity.
-func (k Keeper) IsPreferenceValid(ctx sdk.Context, preferences []types.ValidatorPreference) bool {
+func (k Keeper) IsPreferenceValid(ctx sdk.Context, preferences []types.ValidatorPreference) ([]types.ValidatorPreference, bool) {
+	var weightsRoundedValPrefList []types.ValidatorPreference
 	for _, val := range preferences {
-		_, _, err := k.GetValidatorInfo(ctx, val.ValOperAddress)
+		// round up weights
+		valWeightStr, err := k.RoundValidatorWeights(ctx, val.Weight)
+		if err != nil {
+			return nil, false
+		}
+
+		_, _, err = k.GetValidatorInfo(ctx, val.ValOperAddress)
 		if err != nil {
-			return false
+			return nil, false
 		}
+
+		weightsRoundedValPrefList = append(weightsRoundedValPrefList, types.ValidatorPreference{
+			ValOperAddress: val.ValOperAddress,
+			Weight:         sdk.MustNewDecFromStr(valWeightStr),
+		})
 	}
-	return true
+
+	return weightsRoundedValPrefList, true
+}
+
+// RoundValidatorWeights rounds to 2 digit after the decimal. For ex: 0.999 = 1.0, 0.874 = 0.87, 0.5123 = 0.51
+func (k Keeper) RoundValidatorWeights(ctx sdk.Context, totalWeight sdk.Dec) (string, error) {
+	totalWeightFloat64, err := totalWeight.Float64()
```

Comment:
errr are we OK with using floats here? I'm very very weary of using arch-dependent floating point numbers in the state-machine. The `Dec` types has rounding functions that I believe do something similar to what you're trying to achieve with this method. Have you explored those?

Reply:
checking right now

Reply:
Replaced with SigFigRound!